src/utils/errors.ts

```diff
@@ -113,6 +113,19 @@ export class UnknownCallError extends ElementCallError {
   }
 }
 
+export class FailToGetOpenIdToken extends ElementCallError {
+  public constructor(error: Error) {
+    super(
+      t("error.generic"),
+      ErrorCode.OPEN_ID_ERROR,
+      ErrorCategory.CONFIGURATION_ISSUE,
```

Comment:
I just noticed that this is always `CONFIGURATION_ISSUE`, even when the error is due to network connectivity. Is this something you'd like to fix?

Reply:
> I just noticed that this is always `CONFIGURATION_ISSUE`, even when the error is due to network connectivity. Is this something you'd like to fix?

I added retry for the `request_token` with the built in backoff. So the transiant connectivity problems should be handled properly. I believe that we would need something else that detects when you are offline and handles that globally, WDYT?

There is not much info about the possible failures for `request_token` in the [spec](https://spec.matrix.org/v1.13/client-server-api/#post_matrixclientv3useruseridopenidrequest_token), it only specify 429 error :/
The category for that is up for discussion though, it looked as a sensible default